distribution/lib/Standard/Base/0.0.0-dev/src/Data/List.enso

```diff
@@ -209,9 +209,18 @@ type List
     map self f = case self of
         Nil -> Nil
         Cons h t ->
-            res = Cons (f h) Nil
-            map_helper t res f
-            res
+            go : List -> Any -> (Any -> Any) -> Nothing
+            go list fill = case list of
+                Nil -> fill Nil
+                Cons h t ->
+                    v = f h
+                    res = Meta.lazy_atom (e -> Cons v e)
```

Comment:
this should be a special atom, not a vector – `res.at 0` and `res.at 1` are very unreadable

Reply:
I.e. I want to go `res.value` and `res.fill` instead of `at 0` and `at 1`

Reply:
I don't want to introduce new type just for this purpose, but since [9222ff5](https://github.com/enso-org/enso/pull/4023/commits/9222ff54c94035c7857eaba0fc8a3e5713cb0572) I am reusing `Pair`.